<?xml version="1.0"?>
<launch>
	<!-- acquisition arguments -->
	<arg name="plugin" default='rosneuro::EGDDevice'/>
	<arg name="devarg" default='$(env HOME)/Neurorobotics/gdf_files/d6.gdf'/>
	
	<arg name="framerate" default='16'/>
	<arg name="samplerate" default='512'/>

	<!-- load filter chain parameters -->
	<rosparam command="load" file="$(find rosneuro_dl)/config/ChainCfg.yaml" />
	<!-- load model parameters-->
	<rosparam command="load" file="$(find rosneuro_dl)/config/Model.yaml" />

	<node name="acquisition" pkg="rosneuro_acquisition" type="acquisition" output="screen">
		<param name="~plugin" 	  value="$(arg plugin)"/>
		<param name="~devarg" 	  value="$(arg devarg)"/>
		<param name="~samplerate" value="$(arg samplerate)"/>
		<param name="~framerate"  value="$(arg framerate)"/>
	</node>

	<node name="filterchain_node" pkg="rosneuro_dl" type="filter_chain.py" output="screen" >
		<param name="configname" value="ChainCfg" />
	</node>
	<node name="predict_neural" pkg="rosneuro_dl" type="predict_neural.py" output="screen" >
		<param name="configname" value="Model" />
	</node>


</launch>
